src/gleam/os.gleam

```diff
@@ -0,0 +1,45 @@
+//// Function to interact with the host operating system.
+
+import gleam/list
+import gleam/string
+
+// Internal type for erlang interop.
+external type CharList
+
+external fn os_getenv() -> List(CharList) =
+  "os" "getenv"
+
+external fn os_putenv(key: CharList, value: CharList) -> Bool =
+  "os" "putenv"
+
+external fn os_unsetenv(key: CharList) -> Bool =
+  "os" "unsetenv"
+
+external fn char_list_to_string(CharList) -> String =
+  "erlang" "list_to_binary"
+
+external fn string_to_char_list(String) -> CharList =
+  "erlang" "binary_to_list"
+
+/// Return all environment variables set on the system.
+pub fn get_env() -> List(tuple(String, String)) {
+  list.map(
+    os_getenv(),
+    fn(char_list) {
+      assert Ok(value) = string.split_once(char_list_to_string(char_list), "=")
+      value
+    },
+  )
+}
+
+/// Set an environment variable.
+pub fn put_env(key: String, value: String) -> Nil {
+  let True = os_putenv(string_to_char_list(key), string_to_char_list(value))
```

Comment:
`True` is a partial pattern so this should be `assert`. Or we could not assign it at all?

Reply:
So the only return value from this function is `true`. https://erlang.org/doc/man/os.html#putenv-2

I always try to avoid lines which throw away a return value, but there is no use in it here.
I would vote for assert, docs say it should never fail but we will get a crash if it ever changes.
not assigning at all would be a silent error if the erlang API was ever to change

Reply:
That is true but the return type of that function is Bool, so Gleam doesn't know that the False clause isn't used. Later this code will fail to compile due to the True case not being handled. A safer approach would be to define a `TrueOnly` type that only has a `True` constructor.

We have tests that check the behaviour, and our implementation doesn't care about the return value, so I don't think we need to check what it is.